### PR TITLE
feat/vite storybook 7

### PR DESCRIPTION
**Pull request by ooystein**
* Build with vite and esbuild
* Update to storybook 7-beta (full release coming in march)
* Use vite for building storybook

To simplify setup even further we should consider moving tests from jest to vitest. Then we can get rid of babel and jest configurations and dependencies. If vitest requires just as much configuration it might not be worth it, but if we can use vite for the tests and everything is configured the same it would make the whole setup much simpler to maintain.

### CODE REVIEW

**Comment by ooystein** [comment]
I have tested to link pacakge localy to compare the new build and it seems to work fine. But I encourage reviewers to test this PR against existing projects dependent on this library.

File: vite.config.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+export const getPeerDeps = pkg => Object.keys(pkg.peerDependencies || {})
+
+// https://vitejs.dev/config/
+export default defineConfig({
+  build: {
+    lib: {
+      entry: 'src/index.ts',
+      formats: ['es', 'cjs', 'umd'],
+      name: 'esvintersection',
+      fileName: 'index',
```
Comment (ovineq) [inline vite.config.ts:15]: image
The default file endings got my eyebrow moving. Could we instead use the conventional file endings for the different formats?
```suggestion
      fileName: (format) => {
        const fileName = 'index'
        switch(format) {
          case 'es':
            return `${fileName}.mjs`
          case 'cjs':
            return `${fileName}.cjs`
          case 'umd':
            return `${fileName}.umd.js`
          default:
            return `${fileName}.js`
        }
      },
```
```package.json
  "exports": {
    "import": "./dist/index.mjs",
    "require": "./dist/index.cjs",
    "browser": "./dist/index.umd.js"
  },
```

**Review by ovineq — CHANGES_REQUESTED** [review]
Great work 🥳 
I'm liking the new categorization of the stories in v7. And Vite makes it is so ~blazingly~ fast!
Just a few comments:

**Comment by ovineq** [comment]
> I have tested to link pacakge localy to compare the new build and it seems to work fine. But I encourage reviewers to test this PR against existing projects dependent on this library.

Also tested linking these changes for our usage. Seems to work just fine 🎉

File: .github/workflows/github-pages.yml [hunk 2]
```diff
@@ -31,7 +31,7 @@ jobs:
       run: mkdir docs_out
 
     - name: Generate storybook
-      run:  npx build-storybook -c .storybook -o docs_out/storybook/${{ github.ref.name }}
+      run: npm run storybook build -c .storybook -o docs_out/storybook/${{ github.ref.name }}
```
[the remark this answers is not in the document]
Comment (ooystein) [reply .github/workflows/github-pages.yml:34]: very good catch!

**Comment by ooystein** [comment]
Fixed review comments

**Review by ovineq — APPROVED** [review]
:shipit:

**Pull request merged**